woocommerce/woocommerce-ios
# [Free Trial] Show free trial banner in all top level views (except one)

Part of #9161 

# Why

Previously the free trial banner was only shown in the Dashboard, this PR takes care of showing the Free trial banner in all top-level views (Dashboard, Orders, Products) except the Hub Menu.

The free trial banner is not shown in the hub menu because it has some weird layout issues after navigating to it a few times. I'm assuming this is one of those SwiftUI "uniqueness" that I can't seem to solve.

I have decided that not showing the banner in the hub menu is not a big deal, since the menu has a badge indicating the plan name.

# How

- There is a new type called `FreeTrialBannerPresenter` that is in charge of presenting and dismissing the banner. This is done by listening to the shared `storePlanSynchronizer` state.

- Update (Dashboard, Orders, Products) to include the `FreeTrialBannerPresenter` in their dependencies.

# Demo

https://user-images.githubusercontent.com/562080/233545307-485627c6-1796-47da-a2b0-611d7f344002.mov

# Testing Steps

- Login into a free trial store
- Go through all tabs
- Make sure the free trial banner is present in Dashboard, Orders, and Products Tab
- Make sure the free trial banner is **NOT** present in the Hub tab.

--
